nodes/src/nodes/library/ssrf_protection.py

```
@@ -0,0 +1,288 @@
+    if isinstance(addr, ipaddress.IPv6Address) and addr.ipv4_mapped:
+        check_addrs.append(addr.ipv4_mapped)
+
+    for check_addr in check_addrs:
+        if not _is_blocked(check_addr):
+            continue
+
+        # Check if the address is in the allowlist
+        if any(check_addr in net for net in allow_nets):
+            continue
+
+        raise SSRFError(f'SSRF protection: request to {hostname!r} blocked — resolved IP {check_addr} is in a private/reserved range. If this is intentional, add the IP or CIDR to the ROCKETRIDE_SSRF_ALLOWLIST environment variable or the node-level allowlist.')
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)